### ISSUE TITLE
HonorLt - Strategist nonce replay

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2022-10-astaria-judging#250]
HonorLt

medium

# Strategist nonce replay

## Summary
Strategist's nonce is never automatically incremented thus making it possible to execute the same data multiple times.

## Vulnerability Detail
When validating the commitment, it queries the ```strategistNonce```:
```solidity
    bytes32 hash = keccak256(
      abi.encode(
        STRATEGY_TYPEHASH,
        IAstariaRouter(ROUTER()).strategistNonce(strategy.strategist),
        strategy.deadline,
        root
      )
    );
```
Usually, in such cases, the nonce is automatically incremented afterward but here it is not and has to be manually readjusted (```incrementNonce```). This makes it possible to reuse the same signature multiple times.

## Impact
I did not have enough time to investigate or ask about the intentions here, so sorry if my assumptions are wrong but I believe a nonce is a number that should be treated as valid only once. Now it makes it possible to commit or buyout liens with the same parameters multiple times.

## Code Snippet

https://github.com/sherlock-audit/2022-10-astaria/blob/main/src/VaultImplementation.sol#L93-L107

## Tool used

Manual Review

## Recommendation
Nonce could be automatically incremented after being used.

Duplicate of #59
